src/pages/index.js

```diff
@@ -15,64 +17,177 @@ const cardFormElement = document.querySelector("#add-card-form");
 //Buttons and other DOM nodes
 const profileEditButton = document.querySelector(".profile__edit-button");
 const newCardButton = document.querySelector(".profile__add-button");
+const confirmButton = document.querySelector("#confirm-modal");
 
 //Form data
 const nameInput = profileEditModal.querySelector("#profile-title");
 const jobInput = profileEditModal.querySelector("#description");
 
-//profile form
-const userInfo = new UserInfo(".profile__title", ".profile__description");
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "9a1167a0-01e6-4e49-b3c5-e813c03c355c",
+    "Content-Type": "application/json",
+  },
+});
+
+//information from server
+
+let cardList;
+let userInfo;
+
+api
+  .loadPageResults()
+  .then(([cards, userData]) => {
+    console.log(cards);
+    console.log(userData);
+    cardList = new Section(
+      {
+        items: cards,
+        renderer: createCard,
+      },
+      ".gallery__cards"
+    );
+    cardList.renderItems();
+    userInfo = new UserInfo(
+      ".profile__title",
+      ".profile__description",
+      ".profile__image"
+    );
+    userInfo.setUserInfo(userData);
+  })
+  .catch((err) => console.error(err));
+
+//Render Loading
+function renderLoading(isLoading) {
+  if (isLoading) {
+  }
+}
+
+//edit profile information
 
 function handleProfileFormSubmit(userData) {
-  profilePopupForm.close();
-  userInfo.setUserInfo(userData);
+  profilePopupForm.setLoading(true, "Saving...");
```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```